index.d.ts

```diff
@@ -42,7 +48,9 @@ export interface BaseMockApplication<T, C> extends Application { // tslint:disbl
   /**
    * http request helper
    */
-  httpRequest(): any;
+  httpRequest(): {
+    [key: string]: (url: string) => EggTest;
```

Comment:
```typescript
type Methods = 'get' | 'post' | 'delete' | 'put' | 'head' | 'optons' | 'patch' | 'trace' | 'connect';

...
  httpRequest(): {
    [key in Methods]: (url: string) => EggTest;
  };
...

```
It may be better?

Reply:
How about:

```typescript
type Methods = 'get' | 'post' | 'delete' | 'put' | 'head' | 'optons' | 'patch' | 'trace' | 'connect';

...
  httpRequest(): {
    [key in Methods]: (url: string) => EggTest;
  } & {
    [key: string]: (url: string) => EggTest;
  };
...
```

Reply:
Why? All methods has listed in `Methods`

Reply:
There is commonly used in HTTP, but is not all methods in HTTP.

HTTP request methods list at: https://www.iana.org/assignments/http-methods/http-methods.xhtml.

You can aslo list by `require('http').METHODS`, it looks more than in the list above.

We does not need to list all methods here, so we can list commonly methods and accept any string for other methods.

Reply:
All right.